src/CoreGraphics/CGColor.cs

```diff
@@ -142,12 +139,11 @@ public CGColor (CGColorSpace colorspace, CGPattern pattern, nfloat [] components
 				throw new ArgumentNullException ("colorspace");
 			if (colorspace.handle == IntPtr.Zero)
 				throw new ObjectDisposedException ("colorspace");
-			if (pattern == null)
-				throw new ArgumentNullException ("pattern");
 			if (components == null)
 				throw new ArgumentNullException ("components");
+			var handleof_pattern = pattern.GetNonNullHandle ("pattern");
```

Comment:
```suggestion
			var handleof_pattern = pattern.GetNonNullHandle (nameof (pattern));
```

Reply:
I think `pattern_handle` sounds slightly better than `handleof_pattern`, but it's not a big deal.

Reply:
yeah, it's not meant to be committed as-is, e.g. `nameof`

for `handleof_` prefix it's more for generated code (maybe with a few extra `_` prefixed) as the `*_handle` has an higher probability of already existing and clashing. Or maybe it will end up with `__handle_p1` for the handle of argument 1...

Funnily it's named after `nameof`, hence the lack of `_` inside it ;-)